src/txkube/testing/test/test_matchers.py

```diff
@@ -53,8 +55,8 @@ def test_mismatch(self):
             mismatch.describe(),
             Equals(
                 u"type mismatch:\n"
-                u"reference = <type 'int'> (0)\n"
-                u"actual    = <type 'dict'> ({0: 1})\n"
+                u"reference = " + repr(int) + u" (0)\n"
+                u"actual    = " + repr(dict) + u" ({0: 1})\n"
```

Comment:
I prefer the previous form of this test (though I see why you changed it) because it puts the exact string right into the test.  Maintainers can see that exact string and think about whether it is meaningful or not.

When the string is computed, instead, no one necessarily sees what it is computed as.  Perhaps something meaningless is computed.  In that case, the test is pretty meaningless too.

I'd like to see two versions of the assertion here, one for Python 2 and one for Python 3.  Alternatively, perhaps the implementation should change not depend on Python version for the result.  We don't actually need the Python type object repr boilerplate, probably?  Just saying `int` or `dict` conveys all of the necessary information.